### PR TITLE
4142 internationalization

### PR DESCRIPTION
**Pull request by pgarrison**
Nearly all of the Aggie interface can now be translated into other languages. Translation files are in `public/angular/translations/locale-*.json` (and for emails in `lib/mailer.js`). Info on how to add new translations is in `contributing.md`. 

Some things that should be translated but can't be (yet)
* The strings `* Any Incident` and `* Without Incident` in the reports page for filtering reports by incident.
* Errors messages that originate on the server
  - These can easily be translated in the same manner as the rest of the strings: just replace the string with a translation id and then add the `"id": "string"` pair to the translation files.
* Tooltips (mainly in trends)

### CODE REVIEW

**Comment by pgarrison** [comment]
@andresfib can you translate the emails to spanish for real (just so we have a second language there as an example)?

File: lib/mailer.js [hunk 1]
```diff
@@ -59,29 +60,55 @@ Mailer.prototype.send = function(subject, body, to, from, callback) {
...
-    body: '<p>To change your Aggie password, <a href="<%= protocol %>://<%= host %>/password_reset/<%= token %>">click here</a> or copy the link below into your browser.</p><p><%= protocol %>://<%= host %>/password_reset/<%= token %></p><p>You are receiving this message because someone requested a new password from <%= host %> for an account registered with this email address. If you did not initiate this request, please ignore this email.</p>'
+  es: {
+    newUser: {
+      subject: 'Bienvenido a Aggie!',
+      body: 'Algo en Español'
+    },
+    forgotPassword: {
+      subject: 'Aggie Contraseña Reiniciar Solicitud',
+      body: '<p>En Español</p>'
+    }
```
Comment (hooverlunch) [inline lib/mailer.js]: Should be in separate file please.

File: public/angular/translations/locale-debug.json [hunk 2]
```diff
@@ -0,0 +1,353 @@
...
+  "Read": "READ",
+  "Unread": "UNREAD",
+  "Flagged": "FLAGGED",
+  "Unflagged": "UNFLAGGED",
+  "Read & Unflagged": "READ & UNFLAGGED",
+  "Total Reports": "TOTAL REPORTS",
+  "Reports/min": "REPORTS/MIN",
+  "Escalated Incidents": "ESCALATED INCIDENTS",
+  "Mark all Unread": "MARK ALL UNREAD"
+}
```
Comment (hooverlunch) [inline public/angular/translations/locale-debug.json]: Perhaps add this to .gitignore and put steps to generate (gulp script) and use in Readme file.

File: test/public.angular.translations.test.js [hunk 3]
```diff
@@ -0,0 +1,95 @@
...
+      if (err) return callback(err);
+      callback(null, _.object(allLanguages));
+    });
+  });
+}
+
+describe('Translations files', function() {
+  it('debug language should contain the union of the other languages',
+     function(done) {
+    getTranslations(function(err, allLanguages) {
```
Comment (hooverlunch) [inline test/public.angular.translations.test.js]: Indent issues here.

Comment (pgarrison) [inline test/public.angular.translations.test.js]: Not a problem as far as I can tell (this diff is outdated but the same thing appears in the new diff), just a little odd-looking.

**Comment by pgarrison** [comment]
Translations for emails still have no tests.

**Comment by pgarrison** [comment]
After poking around a little with debug translations on, I would estimate that around 80% of the strings are translated with this. Hopefully we can gradually the rest of the missing ones as we find them.

**Pull request merged**